Add Indexing Pipeline Information and File Classifier Page (#360)
deepset-ai/haystack-website: docs/latest/components/pipelines.mdx

# Pipelines

To build modern search pipelines, you need two things: powerful [building blocks](/pipeline_nodes/overview) and an easy way to stick them together.
The `Pipeline` class is precisely built for this purpose and enables many search scenarios beyond QA.
The core idea is to build a Directed Acyclic Graph (DAG) where each Node is one building block (Reader, Retriever, Generator).

## Indexing Pipelines vs Querying Pipelines

> Indexing pipelines take files as input. They convert them into Haystack Documents and perform some processing steps. Then they index the Documents into a Document Store.

Reviewer: Can we first add some introductory info, like: Indexing pipelines prepare your files for search. Their main objective is to convert your files into Haystack Documents, which are passages of raw text stored in a Document Store (here a link to the Document in glossary?). Indexing pipelines can also preprocess the files by.. (can we add an example here? if not, let's remove the info about processing).

> The first Node in an indexing pipeline will usually be a [File Convertor](/pipeline_nodes/file_converters) or a FileTypeClassifier and take "File" as its input.

Reviewer 2: Typo: File Converter

> The first Node in an indexing pipeline will usually be a [File Converter](/pipeline_nodes/file_converters) or a FileTypeClassifier and take "File" as its input.

Reviewer: How about adding some more info about what these nodes can do: 
Depending on the file types you have, you can use [FileTypeClassifier](/reference/file-classifier) as the first node in your pipeline. It routes your files to appropriate [File Converters)(/pipeline_nodes/file_converters), which then extract text from your files and cast it into Documents.

> The last Node will always be a Document Store.

Reviewer: ```suggestion
The last Node is always a Document Store.
```

> Here's an example of an indexing pipeline that preprocesses and classifies each document:

Reviewer: Are there any other nodes in between? Preprocessor?

```python
from haystack.pipelines import Pipeline

p = Pipeline()
p.add_node(component=text_converter, name="TextConverter", inputs=["File"])
p.add_node(component=preprocessor, name="PreProcessor", inputs=["TextConverter"])
p.add_node(component=document_classifier, name="DocumentClassifier", inputs=["PreProcessor"])
p.add_node(component=document_store, name="DocumentStore", inputs=["DocumentClassifier"])

p.run(file_paths=["filename.txt"])
```

<div className="max-w-xl bg-yellow-light-theme border-l-8 border-yellow-dark-theme px-6 pt-6 pb-4 my-4 rounded-md dark:bg-yellow-900">

> **Tutorial:** have a look at the [Document Classification at Indexing Tutorial](/tutorials/doc-class-index) for a more detailed example of this kind of pipeline.

Reviewer: ```suggestion
**Tutorial:** For a more detailed example of an indexing pipeline, have a look at the [Document Classification at Indexing Tutorial](/tutorials/doc-class-index).
```

</div>

Querying pipelines perform searches (or any other NLP task) on a set of documents that have been indexed into a Document Store.

> The first Node in a querying pipeline will take "Query" as its input and the documents will most often be found within the Retriever which in turn contains a Document Store.

Reviewer: This is a bit unclear. Can we simplify it? Does a query pipeline always have a retriever? 
Maybe we can say something like: The first node in a querying pipeline always takes query as input. To find the right answer, the piepline uses documents stored in a document store.

> A querying pipeline for Open-Domain QA might look something like this:

Reviewer: Here's an example of what a querying pipeline for open-domain question answering may look like:

## Run a Pipeline

The `run()` function is the single command that triggers the execution of the entire pipeline.

> For query pipelines, you will need to provide `query` as an argument:

Reviewer: ```suggestion
For querying pipelines, provide `query` as an argument:
```

Reviewer: If we're using "querying pipelines", let's be consistent throughout the docs.

``` python
query = "What's the history of Quidditch?"
pipeline.run(query=query)
```

> For indexing pipelines, you will need to provide `file_paths` as an argument:

Reviewer: ```suggestion
For indexing pipelines, provide `file_paths` as an argument:
```
